Example in the README is not working

## Bug description

<!-- A clear and concise description of what the bug is. -->

When copy/pasting the example from the readme into a test project, it does not work because the dependencies cannot be satisfied. Adjusting the dependencies results in a building issue.

## To Reproduce

<!-- Steps to reproduce the behavior. -->

```bash
cd /tmp
cargo new rtexample
cd rtexample/
code -n .
# Copy/paste the example from the readme
```

At this point, trying to build results in the following error:

```bash
rtexample on  master [?] via 🦀 v1.59.0
❯ cargo build
    Updating crates.io index
error: failed to select a version for `reqwest-tracing`.
    ... required by package `rtexample v0.1.0 (/private/tmp/rtexample)`
versions that meet the requirements `^0.1.2` are: 0.1.3, 0.1.2

the package `rtexample` depends on `reqwest-tracing`, with features: `opentelemetry_0_17` but `reqwest-tracing` does not have these features.


failed to select a version for `reqwest-tracing` which could resolve this conflict
```

Adjusting the dependency to:

```toml
reqwest-tracing = { version = "0.2.1", features = ["opentelemetry_0_17"] }
```

then produces the following build error:

```bash
error[E0277]: the trait bound `opentelemetry::sdk::trace::Tracer: opentelemetry::trace::tracer::Tracer` is not satisfied
   --> src/main.rs:10:62
    |
10  |   let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
    |                                                  ----------- ^^^^^^ the trait `opentelemetry::trace::tracer::Tracer` is not implemented for `opentelemetry::sdk::trace::Tracer`
    |                                                  |
    |                                                  required by a bound introduced by this call
    |
note: required by a bound in `OpenTelemetryLayer::<S, T>::with_tracer`
   --> /Users/rgreinhofer/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-opentelemetry-0.15.0/src/layer.rs:308:17
    |
308 |         Tracer: otel::Tracer + PreSampledTracer + 'static,
    |                 ^^^^^^^^^^^^ required by this bound in `OpenTelemetryLayer::<S, T>::with_tracer`

error[E0277]: the trait bound `opentelemetry::sdk::trace::Tracer: PreSampledTracer` is not satisfied
   --> src/main.rs:10:62
    |
10  |   let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
    |                                                  ----------- ^^^^^^ the trait `PreSampledTracer` is not implemented for `opentelemetry::sdk::trace::Tracer`
    |                                                  |
    |                                                  required by a bound introduced by this call
    |

[...] 
```

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->
The example in the readme should work right away to allow new users to quickly experience the library.

Adding an example that could be ran with `cargo run --example` and tested via the CI would be ideal.

## Environment

<!-- Please fill the following information. -->

- OS: macOS 12.2.1 (21D62)
- Rust version1.59.0

--- a/reqwest-tracing/README.md
+++ b/reqwest-tracing/README.md
@@ -20,11 +20,11 @@ opentelemetry = "0.17"
 reqwest = "0.11"
 reqwest-middleware = "0.1.1"
 reqwest-retry = "0.1.1"
-reqwest-tracing = { version = "0.1.2", features = ["opentelemetry_0_17"] }
+reqwest-tracing = { version = "0.2.1", features = ["opentelemetry_0_17"] }
 tokio = { version = "1.12.0", features = ["macros", "rt-multi-thread"] }
 tracing = "0.1"
-tracing-opentelemetry = "0.15"
-tracing-subscriber = "0.2"
+tracing-opentelemetry = "0.17"
+tracing-subscriber = "0.3"
 ```
 
 ```rust,skip

